Test - added styles for header

DanielPopchenko opened the pull request:
Please, look at new InputWithSearch component, and at fixed (width) Button component as well.

DanielPopchenko committed:
Test - added styles for header

DanielPopchenko committed:
Addde types for header

DanielPopchenko committed:
Created default expo app

DanielPopchenko committed:
Cleared some files and added the InputWithSearch component.

DanielPopchenko committed:
Modified some files and added types file for Accordion.

olexandermushtyn commented on src/components/Button.tsx:
```diff
@@ -27,7 +27,7 @@ export const Button = ({
   isRounded,
   isIconOnly,
-  isFullWidth,
+  // isFullWidth,
```
Please, remove unnecessary props

olexandermushtyn commented on src/components/Button.tsx:
```diff
@@ -97,7 +97,7 @@ export const Button = ({
       color: NEUTRAL_BLACK_900,
       backgroundColor: NEUTRAL_WHITE_50,
-      minWidth: 322,
+      // minWidth: 322,
```
Always try to remove unnecessary commented code

olexandermushtyn commented on src/components/InputWithSearch.tsx:
```diff
@@ -0,0 +1,60 @@
+        </View>
+      </TouchableOpacity>
+
+      {isShowOptions ? (
```
I think this prop is redundant, make this prop optional and set the default value as "true"

DanielPopchenko committed:
minor changes

DanielPopchenko committed:
Modified inputWithSearch & button, added some properties & styles

DanielPopchenko commented:
At the last commit:
1) added dynamic fontSize property for Button.
2) Fixed issue with inputWithSearch typing something inside.
3) Fixed fontWeight properties.
4) added types for onChangeText, onBlur and onFocus inside input components
5) fixed issue with iconOnly button. Moved fullWidth styles up, so they can be overwritten when isIconOnly property is applied

DanielPopchenko committed:
Changes in InputWithSearch component

olexandermushtyn commented on src/components/InputWithSearch.tsx:
```diff
@@ -0,0 +1,91 @@
+
+  useEffect(() => {
+    setData(options);
+  }, []);
```
Add options inside useEffect dependencies

olexandermushtyn commented on src/components/InputWithSearch.tsx:
```diff
@@ -0,0 +1,91 @@
+  searchValue,
+  placeholder,
+  options,
+  // onSearch,
```
Don't forget to remove unnecessary commented code

olexandermushtyn commented on src/components/InputWithSearch.tsx:
```diff
@@ -0,0 +1,91 @@
+    setIsShowOptions(false);
+  };
+
+  const contains = (option: { label: string; value: string }, query: string) => {
+    const { label } = option;
+    if (label.toLowerCase().includes(query)) {
+      return true;
+    }
+    return false;
+  };
```
If this function doesn't use any component code (props, function, etc.) you can move this function outside of the component to avoid function recreation on every rerender

DanielPopchenko committed:
removed comments, added dep arr, moved func outside

DanielPopchenko committed:
Fixed input with search component

DanielPopchenko committed:
Added additional properties to components

olexandermushtyn commented on src/components/Accordion.tsx:
```diff
@@ -18,14 +10,15 @@ const AccordionItem = ({
   iconOpen,
   iconClose,
   iconLeft,
+  testId,
```
Please remove this prop, it is not needed, because it may differ depending on the library used for testing and if it is needed, it can be added there with {...props}

olexandermushtyn commented on src/components/Accordion.tsx:
```diff
@@ -45,31 +38,28 @@ const AccordionItem = ({
+  iconClose,
+  iconOpen,
+  testId,
+  additionalStyle,
```
Also, follow the props naming convention, if there are any in the default components, use style instead of additionalStyle

DanielPopchenko committed:
fixed issues

olexandermushtyn commented on src/types/input.types.ts:
```diff
@@ -1,26 +1,41 @@
   value: string;
   placeholder: string;
-  onChangeText: any;
+  onChangeText: React.Dispatch<React.SetStateAction<string>>;
```
Change just to Function, cause the function that will get there is not always the setState function

DanielPopchenko committed:
fixed types

DanielPopchenko committed:
removed comments
